### DISCUSSION TITLE
Alien Current generators push players sideways

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#14839] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

current generators push players to the side, not just in the direction they're supposed to go. this effect is compounded greatly if you make the trigger box larger, to the point where you'll go sideways much faster than in the direction you're supposed to.

if you try to use current generators to do things such as reduce gravity or add wind they automatically force players to either side, which is annoying. you can't just put multiple next to each other, because then the forces add together. and since they're still pushing the player to the side, the player wants to be in the intersections at all times.

in order to properly do it, you'd need a giant semicircle array of them. however they'd need each their own precisely calculated force, because if they all had the same force, gravity would just flip sideways. 
![image](https://github.com/user-attachments/assets/a88900ce-f32f-4195-ae8a-e53178e38c69)

they should probably be forcing people in only 1 direction, not just "away from center"
if it absolutely must stay circular, then maybe allow us to increase the radius of it in editor or at least via modding, to values so insane it may as well be flat.

ragdolling also has some strange behavior. in water currents, ragdolling makes the current push you faster in the direction of the current. this is expected. however in air currents, ragdolling somehow makes you fall faster, even if the current is pointing upwards. quite strange.

### Reproduction steps

1. use alien current generator for mapmaking purposes
2. try to do anything with them 
3. it tries to push the player to either side, not just in 1 direction.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.4.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by jigardi** [comment]
Hmmmm, I ran into this issue recently.

The TriggerComponent holds tremendous potential for modding;  but the always-radial force calculation more or less nullifies it all.

A boolean toggle for linear force calculation(ideally with respect to the components rotation),  would allow for a ream of new content.

Off the top of my head, this alone would enable:

- forcefields
- wind/currents
- artificial-gravity
- pseudo-ocean-surfaces
- propellerwash
- slipstreams
- elevators
- massively increased creature variety ( wakes left behind when they swim, moving as though by waterjets,  creatures with pseudo-grab/push abilities via projected force acting as a harmonizing middleman)

-and probably a ton of other stuff besides.

Edit: Clarity.

**Reply by girlyguppy** [reply]
That is ong. Shame they seem to disregard this (they switched this bug report to ideas and suggestions)

**Comment by JeroonBosch** [comment]
Not sure I entirely get what you're reporting. Do you mean the alien current generators push like this?
![image](https://github.com/user-attachments/assets/24e4e999-e2cc-4de8-8fa7-aa2f158f77e8)


Instead of just this:
![image](https://github.com/user-attachments/assets/aee71180-f206-476f-87e6-7aa4d586cd51)

**Reply by girlyguppy** [reply]
Exactly. For current generation specifically, it isn't that big of a deal, but for other uses, it doesn't work

**Comment by Regalis11** [comment]
This has been addressed in our private repo now, fix coming up in the next update!
